### PR TITLE
sink(ticdc): add tests for enable-tidb-extension and extract the config out

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve? 

None

### What is changed and how it works?

* extract the config out
* add tests for enable-tidb-extension opts
* Define some magic string as constants

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
/cc @amyangfei @3AceShowHand 

Could you please take a look? Thanks! 💚 💙 💜 💛 ❤️

File: cdc/sink/producer/kafka/config.go [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+// CompleteByOpts the kafka configuration.
+func (c *Config) CompleteByOpts(sinkURI *url.URL, replicaConfig *config.ReplicaConfig, opts map[string]string) error {
+	c.BrokerEndpoints = strings.Split(sinkURI.Host, ",")
+	params := sinkURI.Query()
+	s := params.Get("partition-num")
+	if s != "" {
+		a, err := strconv.Atoi(s)
+		if err != nil {
+			return err
```
Comment (amyangfei) [inline cdc/sink/producer/kafka/config.go:77]: Better to use an RFC error, ditto for other parameter's Atoi error.
Besides has this error branch be tested?

File: cdc/sink/producer/kafka/config.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+		replicaConfig.Sink.Protocol = s
+	}
+
+	s = params.Get("enable-tidb-extension")
+	if s != "" {
+		_, err := strconv.ParseBool(s)
+		if err != nil {
+			return err
+		}
+		if replicaConfig.Sink.Protocol != "canal-json" {
```
Comment (amyangfei) [inline cdc/sink/producer/kafka/config.go:171]: Should we check the boolean value of `enable-tidb-extension` is true here?

File: cdc/sink/producer/kafka/config.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+	case "lz4":
+		config.Producer.Compression = sarama.CompressionLZ4
+	case "zstd":
+		config.Producer.Compression = sarama.CompressionZSTD
+	default:
+		log.Warn("Unsupported compression algorithm", zap.String("compression", c.Compression))
+		config.Producer.Compression = sarama.CompressionNone
+	}
+
+	// Time out in one minute(120 * 500ms).
```
Comment (amyangfei) [inline cdc/sink/producer/kafka/config.go:270]: Curious about, if each retry reaches the timeout (20s defined in L273), will the total retry time be 120 * 20s?

File: cdc/sink/producer/kafka/config.go (hunk 2 above)
Comment (hi-rustin) [reply cdc/sink/producer/kafka/config.go:171]: It shouldn't be necessary, it should only be possible to configure this value in this protocol.

File: cdc/sink/producer/kafka/config.go (hunk 3 above)
Comment (hi-rustin) [reply cdc/sink/producer/kafka/config.go:270]: It should be a total of 20s.

Comment (hi-rustin) [reply cdc/sink/producer/kafka/config.go:270]: I'll reset it.

File: cdc/sink/producer/kafka/config.go (hunk 2 above)
Comment (amyangfei) [reply cdc/sink/producer/kafka/config.go:171]: I mean user configure sink-uri with `enable-tidb-extension=false&protocol=canal` or other protocols except for `canal-json`. Then the check will fail here.

Comment (amyangfei) [reply cdc/sink/producer/kafka/config.go:171]: althrough `enable-tidb-extension=false` is useless.

File: cdc/sink/producer/kafka/config.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sink/producer/kafka/config.go:77]: Wraped in https://github.com/pingcap/ticdc/blob/77029891ff124a5a0e07d5b4dbd4c4f40c6c6268/cdc/sink/mq.go#L384

Comment (hi-rustin) [reply cdc/sink/producer/kafka/config.go:77]: > Besides has this error branch be tested?

Let me test it.

**Review by amyangfei — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**